Composer/packages/client/src/components/CreationFlow/v2/CreateOptions.tsx

```diff
@@ -0,0 +1,268 @@
+    },
+  };
+};
+
+const rowTitle = (disabled) => {
+  return {
+    cellTitle: {
+      color: disabled ? NeutralColors.gray80 : NeutralColors.black,
+      selectors: {
+        ':hover': {
+          background: disabled ? NeutralColors.white : NeutralColors.gray30,
+          color: disabled ? NeutralColors.gray80 : NeutralColors.black,
+        },
+      },
+    },
+  };
+};
+
+const tableCell = css`
+  outline: none;
+  :focus {
+    outline: rgb(102, 102, 102) solid 1px;
+  }
+`;
+
+const content = css`
+  outline: none;
+`;
+
+const optionKeys = {
+  createFromScratch: 'createFromScratch',
+  createFromQnA: 'createFromQnA',
+  createFromTemplate: 'createFromTemplate',
+};
+
+// -------------------- CreateOptions -------------------- //
+type CreateOptionsProps = {
+  templates: BotTemplate[];
+  onDismiss: () => void;
+  onNext: (data: string) => void;
+  fetchTemplates: (feedUrls?: string[]) => Promise<void>;
+} & RouteComponentProps<{}>;
+
+export function CreateOptionsV2(props: CreateOptionsProps) {
+  const [option] = useState(optionKeys.createFromTemplate);
+  const [disabled] = useState(false);
+  const { templates, onDismiss, onNext } = props;
+  const [templateReadMe, setTemplateReadMe] = useState('');
+  const [currentTemplate, setCurrentTemplate] = useState('');
+  const [emptyBotKey, setEmptyBotKey] = useState('');
+  const [selectedFeed, setSelectedFeed] = useState<{ props: IPivotItemProps }>({ props: { itemKey: csharpFeedKey } });
+
+  const selection = useMemo(() => {
+    return new Selection({
+      onSelectionChanged: () => {
+        const t = selection.getSelection()[0] as BotTemplate;
+        if (t) {
+          setCurrentTemplate(t.id);
+        }
+      },
+    });
+  }, []);
+
+  const handleJumpToNext = () => {
+    let routeToTemplate = emptyBotKey;
+    if (option === optionKeys.createFromTemplate) {
+      routeToTemplate = currentTemplate;
+    }
+
+    if (option === optionKeys.createFromQnA) {
+      routeToTemplate = QnABotTemplateId;
+    }
+
+    if (props.location && props.location.search) {
+      routeToTemplate += props.location.search;
+    }
+
+    onNext(routeToTemplate);
+  };
+
+  const tableColums = [
+    {
+      key: 'name',
+      name: formatMessage('Name'),
+      fieldName: 'name',
+      minWidth: 150,
+      maxWidth: 200,
+      isResizable: !disabled,
+      data: 'string',
+      styles: rowTitle(disabled),
+      onRender: (item) => (
+        <div data-is-focusable css={tableCell}>
+          <div css={content} tabIndex={-1}>
+            {item.name}
+          </div>
+        </div>
+      ),
+    },
+  ];
+
+  const onRenderRow = (props) => {
+    if (props) {
+      return (
+        <DetailsRow {...props} data-testid={props.item.id} styles={rowDetails(disabled)} tabIndex={props.itemIndex} />
+      );
+    }
+    return null;
+  };
+
+  const fetchReadMe = async (moduleName: string) => {
+    try {
+      const response = await httpClient.post(`assets/templateReadme`, {
```

Comment:
We're breaking the layering here by fetching data directly in the view.

Also, why is this HTTP method a post and not a get?

Reply:
I thought it would be appropriate to have this data being fetched at the component level since it is not used anywhere else. Would you have it live in the app state?

Reply:
I agree that it should be in app state, ill add it back there

Reply:
¯\_(ツ)_/¯

Reply:
Its like the whole division of business logic and presentational logic.

Ive also been on teams that ban local state in favor of central. Like even if a component is the only one that needs that data it may not be the case down the line so always place state in a central repo like recoil or redux. 

Thats a little overkill imo but I do like the division of business logic and presentational logic aspect.

Reply:
Actually I am going to leave this as is. We need to settle the ideal read me/template detail acquisition story. If we are no longer going down the readme grabbing route then it does not make sense to move this to app state since that will potentially be throw away code.